src/main/java/org/veupathdb/service/eda/ds/plugin/FilteredVariableMetadataPlugin.java

```diff
@@ -0,0 +1,128 @@
+import org.veupathdb.service.eda.generated.model.ContinuousVariableMetadataPostRequest;
+import org.veupathdb.service.eda.generated.model.ContinuousVariableMetadataPostResponse;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.veupathdb.service.eda.common.plugin.util.RServeClient.useRConnectionWithRemoteFiles;
+
+public class FilteredVariableMetadataPlugin extends AbstractEmptyComputePlugin<ContinuousVariableMetadataPostRequest, ContinuousVariableMetadataSpec> {
+  
+  private static final Logger LOG = LogManager.getLogger(FilteredVariableMetadataPlugin.class);
+
+  private static final ManagedMap<String,ContinuousVariableMetadataPostResponse> RESULT_CACHE =
+      new ManagedMap<>(5000, 2000);
+
+  private String _cacheKey;
+  private ContinuousVariableMetadataPostResponse _cachedResponse;
+
+  @Override
+  protected Class<ContinuousVariableMetadataPostRequest> getVisualizationRequestClass() {
+    return ContinuousVariableMetadataPostRequest.class;
+  }
+
+  @Override
+  protected Class<ContinuousVariableMetadataSpec> getVisualizationSpecClass() {
+    return ContinuousVariableMetadataSpec.class;
+  }
+
+  @Override
+  public ConstraintSpec getConstraintSpec() {
+    return new ConstraintSpec()
+      .dependencyOrder(List.of("variable"))
+      .pattern()
+        .element("variable")
+          .shapes(APIVariableDataShape.CONTINUOUS)
+          .description("Variable must be continuous.")
+      .done();
+  }
+  
+  @Override
+  protected void validateVisualizationSpec(ContinuousVariableMetadataSpec pluginSpec) throws ValidationException {
+    validateInputs(new DataElementSet()
+      .var("variable", pluginSpec.getVariable()));
+  }
+
+  @Override
+  protected List<StreamSpec> getRequestedStreams(ContinuousVariableMetadataSpec pluginSpec) {
+    String entityId = pluginSpec.getVariable().getEntityId();
+    _cacheKey = EncryptionUtil.md5(
+        JsonUtil.serializeObject(getSubsetFilters()) +
+        "|" + JsonUtil.serializeObject(pluginSpec)
+    );
+    _cachedResponse = RESULT_CACHE.get(_cacheKey);
+    if (_cachedResponse != null) LOG.info("Found cached result.");
+
+    return _cachedResponse != null ? Collections.emptyList() : ListBuilder.asList(
+      new StreamSpec(DEFAULT_SINGLE_STREAM_NAME, pluginSpec.getVariable().getEntityId())
+        .addVar(pluginSpec.getVariable()));
+  }
+
+  @Override
+  protected void writeResults(OutputStream out, Map<String, InputStream> dataStreams) throws IOException {
+    PluginUtil util = getUtil();
+    ContinuousVariableMetadataSpec spec = getPluginSpec();
+    var jsonWrapper = new Object(){ String value = "{"; };
```

Comment:
Probably better to use a StringBuilder here with calls to append() below, but could also just use a JSONObject and build from there, converting to String at the end.